torch_geometric/nn/aggr/base.py

```diff
@@ -100,6 +106,24 @@ def set_validate_args(value: bool):
         """
         Aggregation._validate = value
 
+    @staticmethod
+    @contextmanager
+    def validation_context(value: bool):
```

Comment:
Wondering if we can do this more broadly in PyG, e.g.,
```python
with torch_geometric.validate(False):
    out = model(data.x, data.edge_index)
```

Reply:
Yes I like that idea, we could coalesce other interfaces (`MessagePassing`, others? ) to have an optional validation step that can be toggled off for ahead-of-time compilation.

Reply:
Yeah exactly. How do we wanna proceed with this. Do we still want to merge this PR as it is and do the re-factor later?

Reply:
I'd like feedback on the `validate` module I've added in the current version of this patch and if it looks ok we can merge this.  We can then refactor other interfaces (only `MessagePassing` is the other one I'm currently aware of) to introduce optional data validation.

Reply:
Sounds good, the PR looks good to me :)

Reply:
> We can then refactor other interfaces (only `MessagePassing` is the other one I'm currently aware of) to introduce optional data validation.

I looked more into the `MessagePassing` and from what I can tell all the places where a `ValueError` is raised should be fine in a compiled context.  The main thing that needs to be avoided are places where there are conditional branches that depend on the input data (a good example of this is in the aggregation interface).